### PR TITLE
Fix more SAT findings and add a few suppressions

### PR DESCRIPTION
**Pull request by wborn**
Signed-off-by: Wouter Born <github@maindrain.net>

### CODE REVIEW

File: bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java [hunk 1]
```diff
@@ -134,7 +134,7 @@ public Integer getRank() {
             StateDescription dynamicStateDescription = provider.getStateDescription(channel, originalStateDescription,
                     locale);
             if (dynamicStateDescription != null) {
-                if (dynamicStateDescription == originalStateDescription) {
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java:137]: Can we revert this? There was a reason to use the comparison against the reference here: https://github.com/openhab/openhab-core/pull/1817

File: bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelCommandDescriptionProvider.java [hunk 2]
```diff
@@ -90,7 +90,7 @@ public ChannelCommandDescriptionProvider(final @Reference ItemChannelLinkRegistr
                 CommandDescription dynamicCommandDescription = dynamicCommandDescriptionProvider
                         .getCommandDescription(channel, originalCommandDescription, locale);
                 if (dynamicCommandDescription != null) {
-                    if (dynamicCommandDescription == originalCommandDescription) {
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelCommandDescriptionProvider.java:93]: Can we revert this? There was a reason to use the comparison against the reference here: https://github.com/openhab/openhab-core/pull/1817

**Review by cweitkamp — COMMENTED** [review]
Thanks you. I left some remarks inside. If that does not work we have to check the implementation of the  `equals()` method in both classes to suppress the warning in a proper way.

File: bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java (hunk 1 above)
Comment (wborn) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java:137]: Good catch! I'll restore it and add a comment so we won't run into this regression again. :blush:

**Review by cweitkamp — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**